prediction_market_agent/agents/think_thoroughly_agent/think_thoroughly_agent.py

```diff
@@ -32,8 +46,16 @@ class Scenarios(BaseModel):
 
 
 class CrewAIAgentSubquestions:
+    pinecone_handler: t.Optional[PineconeHandler] = None
+    model: str
+
     def __init__(self, model: str) -> None:
         self.model = model
+        self.subgraph_handler = OmenSubgraphHandler()
+
+    def _init_pinecone(self) -> None:
+        if not self.pinecone_handler:
+            self.pinecone_handler = PineconeHandler()
```

Comment:
Loading it lazily in case the API key not present.

Reply:
Why not just do `self.pinecone_handler = PineconeHandler()` inside of `__init__` ?